nuxeo-retention-web/elements/nuxeo-retention-analytics.js

```diff
@@ -0,0 +1,365 @@
+/**
+(C) Copyright Nuxeo Corp. (http://nuxeo.com/)
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+import moment from '@nuxeo/moment';
+import { mixinBehaviors } from '@polymer/polymer/lib/legacy/class.js';
+import { html } from '@polymer/polymer/lib/utils/html-tag.js';
+import { I18nBehavior } from '@nuxeo/nuxeo-ui-elements/nuxeo-i18n-behavior.js';
+import { FormatBehavior } from '@nuxeo/nuxeo-ui-elements/nuxeo-format-behavior.js';
+import '@nuxeo/nuxeo-dataviz-elements/nuxeo-repository-data.js';
+import '@nuxeo/nuxeo-ui-elements/widgets/nuxeo-card.js';
+import '@nuxeo/chart-elements/chart-bar.js';
+import '@nuxeo/nuxeo-ui-elements/widgets/nuxeo-date-picker.js';
+
+/**
+`nuxeo-retention-analytics`
+@group Nuxeo UI
+@element nuxeo-retention-analytics
+*/
+class RetentionAnalytics extends mixinBehaviors([I18nBehavior, FormatBehavior], Nuxeo.Element) {
+  static get template() {
+    return html`
+      <style include="nuxeo-styles">
+        .infoCard {
+          @apply --layout-horizontal;
+          @apply --layout-justified;
+          @apply --layout-baseline;
+        }
+
+        .rangeSelector {
+          @apply --layout-horizontal;
+          @apply --layout-baseline;
+        }
+
+        .dateLabel {
+          margin: 0 8px;
+        }
+
+        .flex-layout {
+          @apply --layout-wrap;
+          @apply --layout-horizontal;
+          @apply --layout-justified;
+        }
+
+        .flex-layout nuxeo-card {
+          flex: 1 0;
+          text-align: center;
+          width: 32%;
+          min-width: 30rem;
+        }
+
+        nuxeo-card .alignLeft {
+          text-align: start;
+        }
+
+        nuxeo-data-table {
+          height: 550px;
+        }
+
+        chart-bar,
+        chart-pie {
+          margin: 25px auto 0 auto;
+          max-width: 100%;
+          display: block;
+          font-size: 0.8rem;
+        }
+
+        chart-bar {
+          padding: 2em;
+        }
+
+        .mainRetentionInfo {
+          color: #7e90a5;
+          text-transform: uppercase;
+          margin-inline-end: 8px;
+        }
+
+        .retentionCount {
+          font-weight: bold;
+        }
+
+        .chartBar {
+          text-align: center;
+        }
+
+        .retentionStatus {
+          margin-right: 8px;
+          margin-left: 8px;
+        }
+      </style>
+      <nuxeo-page selected-attribute="visible">
```

Comment:
what's this `selected-attribute` for? iirc it's only used in the page container (ie `iron-pages`) to determine what attribute to set on the currently selected item

Reply:
@nelsonsilva i am using it to know if the page is visible for the observer - maybe I understood wrongly how to use it?

Reply:
ah! no, no need to have it here, in the case of analytics the relevant usage is here https://github.com/nuxeo/nuxeo-web-ui/blob/master/elements/nuxeo-admin/nuxeo-analytics.js#L57 this tells the parent `iron-pages` to toggle the `visible` attribute on the currently selected children. This is what causes your `visible` property here to be toggled. `nuxeo-page`doesn't even have a `selectedAtrribute` property.

Reply:
ops! thanks Nelson! changed.